frontend/src/components/AdminDashboard.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch } from 'react-redux'; 
+import { getAllTasksForAdmin, updateTask, deleteTask } from '../services/taskService';
+import { logout } from '../features/auth/authSlice';
+
+function AdminDashboard() {
+  const navigate = useNavigate();
+  const dispatch = useDispatch(); 
+  const [tasks, setTasks] = useState([]);
+  const [totalTasks, setTotalTasks] = useState(0);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [editingTaskId, setEditingTaskId] = useState(null);
+  const [newTaskText, setNewTaskText] = useState('');
+  const tasksPerPage = 10;
+
+  useEffect(() => {
+    fetchTasks(); 
+  }, [currentPage, navigate]);
+
+  const fetchTasks = async () => {
+    try {
+      const data = await getAllTasksForAdmin(currentPage);
+      setTasks(data.data.tasks || []);
+      setTotalTasks(data.data.totalTasks || 0);
+    } catch (error) {
+      console.error(error);
+      navigate('/login'); 
+    }
+  };
+
+  const handleUpdateTask = async (taskId) => {
+    try {
+      await updateTask(taskId, { task: newTaskText });
+      setEditingTaskId(null);
+      setNewTaskText('');
+      fetchTasks(); 
+    } catch (error) {
+      console.log("Error updating task:", error);
```

Comment:
please use tost library for show error or success

Reply:
resolved.